### ISSUE TITLE
Return proper error instances

### ISSUE THREAD

**Issue by Prinzhorn** [komola/paymill-node#7]
It would be great if an instance of `Error` would be returned instead of arbitrary JSON. I feel like it's kind of a convention just like `error` is the first parameter of a callback.

Benefits: we get access to `error.stack` for free, which is essential for debugging.

So basically I want an `Error` instance with `.stack` and `.message` and a property (maybe `meta`) which contains the status code and response (the thing that's currently returned as error).

**Comment by sebastianhoitz** [comment]
I just pushed some changes. Can you take a look at them? If they are ok, I will publish a new version on npm.

**Comment by Prinzhorn** [comment]
I can't find the documentation on how the JSON looks like that Paymill returns.

For example

```
curl https://api.paymill.com/v2/clients -u invaliduser:
```

returns 

```
{"error":"Access Denied","exception":"InvalidAuthentication"}
```

so there's no `response.error.message` etc. I'm not sure if Paymill is inconsistent or if it never contains an `error` object.

**Comment by Prinzhorn** [comment]
Now there are still many cases where you pass a raw string as error, e.g. https://github.com/komola/paymill-node/blob/master/lib/main.js#L114

**Comment by Prinzhorn** [comment]
Hm, this bit me (again) twice in the past two days. The most recent case is that I just have this in the logs

```
Error: Response code is not ok
    at IncomingMessage.<anonymous> (/home/.........../node_modules/paymill-node/lib/main.js:27:21)
    at IncomingMessage.EventEmitter.emit (events.js:117:20)
    at _stream_readable.js:920:16
    at process._tickCallback (node.js:415:13)
```

Is there any way we can map the paymill error response to a useful error message?

**Comment by sebastianhoitz** [comment]
Hey @Prinzhorn I just merged in the pull request #11 which changed how errors are being handled.

I hope this helps with this issue? If not, please re-open.
